### ISSUE TITLE
CMake ctest helper

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan#1148]
Hi!

The awesome feature, cmake helper, made my life easier to write some receipts, but when I need write some test_package conanfile, I have:

```python
    cmake = None

    def build(self):
        self.cmake = CMake(self.settings)
        self.cmake.configure(self, source_dir=self.conanfile_directory, build_dir="./")
        self.cmake.build(self)

    def test(self):
        target_test = "RUN_TESTS" if self.settings.os == "Windows" else "test"
        self.cmake.build(self, target=target_test)
```

I use CTest for my unit tests, so, I put ``enable_testing`` and ``add_test`` in my cmake file, but I need select the target word by platform.  Maybe, would be a better approach if I have:

```python
    cmake = None

    def build(self):
        self.cmake = CMake(self.settings)
        self.cmake.configure(self, source_dir=self.conanfile_directory, build_dir="./")
        self.cmake.build(self)

    def test(self):
        self.cmake.test(self)
```
This helper will solve the target name by platform and call cmake.build.

**Comment by memsharded** [comment]
Are the targets for test always named the same? (RUN_TESTS and test)
This awesome feature is work by @sixten-hilborn, love to hear his opinion. 
This feature would be very simple, why don't you try it yourself @uilianries? :)

**Comment by uilianries** [comment]
By default, CMake apply ``test`` for Linux, or ``RUN_TESTS`` for Visual Studio, as target, when ``enable_testing()`` is present in CMakeFile.txt.  
Here is the explanation: [Testing_With_CTest](https://cmake.org/Wiki/CMake/Testing_With_CTest)

However, anyone can create a custom target instead to run CTest:
```cmake
# CMakeLists.txt
add_executable(test-app main.cpp)
add_custom_target(run-all-tests
                  WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
                  COMMAND test-app)
add_dependecies(run-all-tests test-app)                                
```
And run:
```shell
$ cmake --build . --target run-all-tests
```
In this case, the better choice is just use ``cmake.build(target="run-all-tests")``.

Great idea, I can implement this little point :)

**Comment by sixten-hilborn** [comment]
I like this idea :) Peeked at your changes, looks good to me!

**Comment by memsharded** [comment]
Hi,

Some housekeeping: I think this was contributed and released some time ago, right?. Now the ``CMake`` helper has a ``test()`` method., not sure why this is open yet, was this the goal of this issue, right? Closing the issue, but please reopen if I am wrong.

Thanks!

**Comment by uilianries** [comment]
Right, was merged in Conan 0.22.

Regards.
